### PR TITLE
ci: create workflow to release c8 optimize

### PR DESCRIPTION
**Pull request by misiekhardcore**
## Description

This PR contains the workflow file to release C8 Optimize from the monorepo All the other changes to the release related files were done in #19666 as the brranch related to that PR was used as a base branch for the release and all the release files are read from that branch

### CODE REVIEW

File: .github/workflows/optimize-release-optimize-c8-only.yml [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+            echo "REVISION=${{ steps.define-values.outputs.git_commit_hash }}"
+            echo "MAJOR_OR_MINOR=${{ steps.release-type.outputs.is_major_or_minor }}"
+            echo "IS_PATCH=${{ steps.release-type.outputs.is_patch }}"
+            echo "IS_ALPHA=${{ steps.release-type.outputs.is_alpha }}"
+            echo "IS_RC=${{ steps.release-type.outputs.is_rc }}"
+            echo "IS_DRY_RUN=${{ github.event.inputs.IS_DRY_RUN || 'false' }}"
+            echo "TAG=${{ github.event.inputs.RELEASE_VERSION || '8.6.0-alpha3-rc3' }}-optimize-test"
+          } >> "$GITHUB_ENV"
+
+      - name: Login to prod Harbor docker registry
```
Comment (misiekhardcore) [inline .github/workflows/optimize-release-optimize-c8-only.yml:134]: Do we need this?

Comment (OmranAbazid) [reply .github/workflows/optimize-release-optimize-c8-only.yml:135]: I believe this is used in the smoke test to pull the docker image

**Review by OmranAbazid — COMMENTED** [review]
See comments

File: .github/workflows/optimize-release-optimize-c8-only.yml (hunk 1 above)
Comment (misiekhardcore) [reply .github/workflows/optimize-release-optimize-c8-only.yml:135]: ah, alright! that makes sense

File: .github/workflows/optimize-release-optimize-c8-only.yml [hunk 2]
```diff
@@ -173,98 +145,159 @@ jobs:
...
-            # ^ starting ~2023-10-10, the hostname resolution for the cambpm docker-compose service
-            # http://cambpm:8080 does not work anymore for neither SaaS nor self-hosted Linux runners.
-            # 172.17.0.1 is the host IP address from the container perspective, and cambpm is running
-            # exposed there on port 8080.
+          ZEEBE_VERSION: ${{ steps.pom-info.outputs.x_zeebe_version }}
+          IDENTITY_VERSION: ${{ steps.pom-info.outputs.x_identity_version }}
+      # ^ starting ~2023-10-10, the hostname resolution for the cambpm docker-compose service
+      # http://cambpm:8080 does not work anymore for neither SaaS nor self-hosted Linux runners.
+      # 172.17.0.1 is the host IP address from the container perspective, and cambpm is running
+      # exposed there on port 8080.
```
[the remark this answers is not in the document]
Comment (misiekhardcore) [reply .github/workflows/optimize-release-optimize-c8-only.yml:287]: good catch, I will push a fix for this

**Review by buccarel — APPROVED** [review]

**Review by OmranAbazid — APPROVED** [review]
Can you please squash the commits into a single one?
Squash merge is not enabled and it has to be done locally and force pushed.

**Pull request merged**